.travis.yml

```diff
@@ -248,12 +248,12 @@ install: yarn run all:install
 # Only deploy and run integration tests when not a PR
 # to handle TravisCI PR restrictions. See:
 # https://docs.travis-ci.com/user/pull-requests#Pull-Requests-and-Security-Restrictions
-script: >
-  yarn run ci:test &&
-  if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
-    yarn run ci:deploy test &&
-    yarn run ci:test:integration
-  fi
+script:
+  - 'yarn run ci:test'
+  - 'if [ ${TRAVIS_BRANCH} = "master" ]; then
+      yarn run ci:deploy test || travis_terminate 1;
```

Comment:
The if block is executed entirely as one command, and the exit code is checked after it is finished.  travis_terminate 1 fails the build immediately if either of these yarn commands fail